cdap-app-fabric/src/main/java/io/cdap/cdap/config/PreferencesService.java

```diff
@@ -176,101 +190,118 @@ private void deleteConfig(EntityId entityId) {
-  public Map<String, String> getProperties(ProgramId programId) {
+  public Map<String, String> getProperties(ProgramId programId)
+    throws Exception {
     return getConfigProperties(programId);
   }
 
-  public PreferencesDetail getPreferences(ProgramId programId) {
+  public PreferencesDetail getPreferences(ProgramId programId)
+    throws Exception {
     return get(programId);
   }
 
   /**
    * Get instance level resolved preferences
    */
   // TODO: remove and replace callsites with getResolvedPreferences
-  public Map<String, String> getResolvedProperties() {
+  public Map<String, String> getResolvedProperties()
+    throws Exception {
     return getConfigResolvedProperties(new InstanceId(""));
   }
 
-  public PreferencesDetail getResolvedPreferences() {
+  public PreferencesDetail getResolvedPreferences()
+    throws Exception {
     return getResolved(new InstanceId(""));
   }
 
   /**
    * Get namespace level resolved preferences
    */
   // TODO: remove and replace callsites with getResolvedPreferences
-  public Map<String, String> getResolvedProperties(NamespaceId namespaceId) {
+  public Map<String, String> getResolvedProperties(NamespaceId namespaceId)
+    throws Exception {
     return getConfigResolvedProperties(namespaceId);
   }
 
-  public PreferencesDetail getResolvedPreferences(NamespaceId namespaceId) {
+  public PreferencesDetail getResolvedPreferences(NamespaceId namespaceId)
+    throws Exception {
     return getResolved(namespaceId);
   }
 
   /**
    * Get app level resolved preferences
    */
   // TODO: remove and replace callsites with getResolvedPreferences
-  public Map<String, String> getResolvedProperties(ApplicationId appId) {
+  public Map<String, String> getResolvedProperties(ApplicationId appId)
+    throws Exception {
     return getConfigResolvedProperties(appId);
   }
-  public PreferencesDetail getResolvedPreferences(ApplicationId appId) {
+
+  public PreferencesDetail getResolvedPreferences(ApplicationId appId)
+    throws Exception {
     return getResolved(appId);
   }
 
   /**
    * Get program level resolved preferences
    */
   // TODO: remove and replace callsites with getResolvedPreferences
-  public Map<String, String> getResolvedProperties(ProgramId programId) {
-   return getConfigResolvedProperties(programId);
+  public Map<String, String> getResolvedProperties(ProgramId programId)
+    throws Exception {
+    return getConfigResolvedProperties(programId);
   }
-  public PreferencesDetail getResolvedPreferences(ProgramId programId) {
+
+  public PreferencesDetail getResolvedPreferences(ProgramId programId)
+    throws Exception {
     return getResolved(programId);
   }
 
   /**
    * Set instance level preferences
    */
-  public void setProperties(Map<String, String> propMap)
-    throws NotFoundException, ProfileConflictException, BadRequestException {
+  public void setProperties(Map<String, String> propMap) throws Exception {
```

Comment:
Same comment as above, seems like we support a level higher than namespace.

Reply:
Same here. Auth extension can decide to ignore or enforce if it is an instance id